dll/win32/samsrv/samrpc.c

```diff
@@ -439,7 +439,7 @@ SamrQuerySecurityObject(IN SAMPR_HANDLE ObjectHandle,
     }
 
     /* Fill the SD data buffer and return it to the caller */
-    SdData->Length = RelativeSdSize;
+    SdData->Length = ResultSdSize;
```

Comment:
I should point you out to the fact that the same potentially problematic construct is also used in lsasrv:
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/lsasrv/lsarpc.c;hb=88808bad8a29213bc51ca54db265e9425a457981#l257